### PR TITLE
Minor: Add multi ordering test for array agg order

### PR DESCRIPTION
**Pull request by jayzhan211**
## Rationale for this change

I found the reason why we need level 3 vec in `convert_array_agg_to_orderings` https://github.com/apache/arrow-datafusion/pull/7629/files#r1357637090. For multi ordering cases, we have len > 1.

Add multi ordering test to improve the test coverage, also simply the code.

### CODE REVIEW

File: datafusion/physical-expr/src/aggregate/array_agg_ordered.rs [hunk 1]
```diff
@@ -231,13 +231,27 @@ impl Accumulator for OrderSensitiveArrayAggAccumulator {
...
             }
 
-            let orderings = ScalarValue::convert_array_to_scalar_vec(agg_orderings)?;
-            // Ordering requirement expression values for each entry in the ARRAY_AGG list
-            let other_ordering_values = self.convert_array_agg_to_orderings(orderings)?;
-            for v in other_ordering_values.into_iter() {
-                partition_ordering_values.push(v);
+            // Get the inner struct array
+            let ordering_array = agg_orderings.value(0);
+            let orderings = ScalarValue::convert_array_to_scalar_vec(&ordering_array)?
```
Comment (jayzhan211) [inline datafusion/physical-expr/src/aggregate/array_agg_ordered.rs:236]: I think we always have ListArray(StructArray) for `agg_orderings`, so we can safely get `orderings` with `convert_array_to_scalar_vec` only.

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 2]
```diff
@@ -106,6 +106,36 @@ FROM
...
+[5, 6, 7, 8, 9, 1, 2, 3, 4, 10]
+
+query TT
+explain select array_agg(c1 order by c2 desc, c3) from agg_order;
+----
+logical_plan
+Aggregate: groupBy=[[]], aggr=[[ARRAY_AGG(agg_order.c1) ORDER BY [agg_order.c2 DESC NULLS FIRST, agg_order.c3 ASC NULLS LAST]]]
+--TableScan: agg_order projection=[c1, c2, c3]
+physical_plan
+AggregateExec: mode=Final, gby=[], aggr=[ARRAY_AGG(agg_order.c1)]
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/aggregate.slt:132]: AggregateExec mode is final, so run `merge_batch`

File: datafusion/physical-expr/src/aggregate/array_agg_ordered.rs [hunk 3]
```diff
@@ -329,6 +319,8 @@ impl OrderSensitiveArrayAggAccumulator {
             .collect();
         let struct_type = DataType::Struct(Fields::from(fields));
 
+        // TODO: Since ScalarValue::Struct is not yet includes StructArray,
+        // so we need to convert it to ListArray for ScalarValue::List
         let arr = ScalarValue::new_list(&orderings, &struct_type);
         Ok(ScalarValue::List(arr))
```
Comment (jayzhan211) [inline datafusion/physical-expr/src/aggregate/array_agg_ordered.rs:325]: `agg_orderings` is the return value of `evaluate_orderings`

**Review by alamb — APPROVED** [review]
The code looks reasonable to me. Thank you @jayzhan211 

 I wonder if @mustafasrepo could give this a look as I think he contributed the original implementation of this aggregate

**Review by mustafasrepo — APPROVED** [review]
It is LGTM!. Thanks @jayzhan211 for this PR.

**Comment by alamb** [comment]
Thanks again

**Pull request merged**